lib/types/json.js

```diff
@@ -104,12 +108,15 @@ function json (options) {
 
     req.body = req.body || {}
 
+    // BEGIN - Check empty body
+    /* 
```

Comment:
I believe this has been left commented out on accident -- this is how all the parsers work, they only parse bodies and this does not parse when there is no body at all (this is different from a zero-length body, i.e. empty string, which AFAIK is that you are adding an error for).